app/notification_test.go

```diff
@@ -861,22 +861,17 @@ func TestGetExplicitMentionsAtHere(t *testing.T) {
 
 	for message, shouldMention := range cases {
 		post := &model.Post{Message: message}
-		if m := getExplicitMentions(post, nil); m.HereMentioned && !shouldMention {
-			t.Fatalf("shouldn't have mentioned @here with \"%v\"", message)
-		} else if !m.HereMentioned && shouldMention {
-			t.Fatalf("should've mentioned @here with \"%v\"", message)
-		}
+		m := getExplicitMentions(post, nil)
+		require.False(t, m.HereMentioned && !shouldMention, "shouldn't have mentioned @here with \"%v\"")
+		require.False(t, !m.HereMentioned && shouldMention, "should've mentioned @here with \"%v\"")
 	}
 
 	// mentioning @here and someone
 	id := model.NewId()
-	if m := getExplicitMentions(&model.Post{Message: "@here @user @potential"}, map[string][]string{"@user": {id}}); !m.HereMentioned {
-		t.Fatal("should've mentioned @here with \"@here @user\"")
-	} else if len(m.Mentions) != 1 || m.Mentions[id] != KeywordMention {
-		t.Fatal("should've mentioned @user with \"@here @user\"")
-	} else if len(m.OtherPotentialMentions) > 1 {
-		t.Fatal("should've potential mentions for @potential")
-	}
+	m := getExplicitMentions(&model.Post{Message: "@here @user @potential"}, map[string][]string{"@user": {id}})
+	require.True(t, m.HereMentioned, "should've mentioned @here with \"@here @user\"")
+	require.False(t, len(m.Mentions) != 1 || m.Mentions[id] != KeywordMention, "should've mentioned @user with \"@here @user\"")
```

Comment:
I would encourage breaking down anything with a `||` to multiple statements. For example, this one can be broken down into the following:
```suggestion
	require.Len(t, m.Mentions, 1,  "should've mentioned @user with \"@here @user\"")
	require.Equal(t, KeywordMention, m.Mentions[id],  "should've mentioned @user with \"@here @user\"")
```
You can apply this to all your require statements that have a OR statement in them. This helps pinpoint which clause failed to more easily debug.

Also, if you are wondering why I put `KeywordMention` first, it's because as per require Docs:
```
func Equal(t TestingT, expected interface{}, actual interface{}, msgAndArgs ...interface{})
```
The expected value goes first. It's a common mistake that is made to put the actual value first and then the expected.